aiChatterBox/aiChatterBox/MainWindow.xaml.cs

```diff
@@ -0,0 +1,284 @@
+﻿//Sauraav Jayrajh
+//sauraavjayrajh@gmail.com
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+
+namespace aiChatterBox
+{
+    public partial class MainWindow : Window
+    {
+        //Client to Handle API Interactions
+        private static readonly HttpClient client = new HttpClient();
+        //Adress of Ollama Running On Machine, Change if Needed
+        public static string localhost = "http://localhost:11434";
+        //Store Chats
+        private List<List<string>> chats = new List<List<string>>();
+        //Store current chat index
+        private int currentChatIndex = -1;
+        //To Indicate Start/Stop of Loading Animation
+        private CancellationTokenSource currentCancellationTokenSource = null;
+
+        //Colours for Chat Messages
+        public Brush userMessageColour = Brushes.Teal;
+        public Brush llamaMessageColour = Brushes.DarkBlue;
+        public Brush errorMessageColour = Brushes.Crimson;
+
+        //File to Store Chats
+        private string chatsFilePath = "pastChats.json";
+
+        public MainWindow()
+        {
+            InitializeComponent();
+            LoadChatsFromFile();
+        }
+
+        //Reuseable Method to Handle Submission of Prompt
+        public async void submitPrompt()
+        {
```

Comment:
I am a little rusty on WinForms and WPF, which is called UWP application and MAUI, however maybe consider haveing a look at this video [Stop using async void in C#!](https://www.youtube.com/watch?v=ZFWxSQ-KjUc) and see if it applies. Maybe try use:

```
public async Task submitPrompt()
{
}
```
As void returns nothing and Task will return task which allows threads to safely return